drv/riscv-plic-server/build.rs

```diff
@@ -0,0 +1,173 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use indexmap::IndexMap;
+use serde::Deserialize;
+use std::collections::{BTreeMap, HashMap};
+use std::env;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+
+task_config::task_config! {
+    ints: &'static [u16],
+    tasks: &'static [&'static str],
+    notification: &'static [u32],
```

Comment:
I think this is fine. But another way to do this would be to put these three members together and use a vector of that structure. Something like this:

```
struct interrupt_config {
     int: u16,
     task: str,
     notification: u32,
}

task_config::task_config! {
      ints: Vec<interrupt_config>,
      pbits: u8,
}
```
It might things slightly easier to write in app.toml:
```
[tasks.ext_int_ctrl.config]
ints = {{int = 52, task = "rtc", notification = 1}, {int = 77, task = "dummy", notification = 1}}
pbits = 3
```

Anyways, I think we don't need to change anything right now. But if the app.toml starts getting confusing when adding more and more interrupts handled by plic, we can think about revisiting this.

Reply:
I tried doing this, but the macro that parses this expects a very simple input. I can see if that can be changed later.

Reply:
Ah okay. SG.